pyatlan/model/core.py

```diff
@@ -87,21 +84,26 @@ def __eq__(self, other):
         )
 
     @classmethod
-    def _convert_to_display_text(cls, data):
-        from pyatlan.cache.atlan_tag_cache import AtlanTagCache
-
+    def _convert_to_display_text(
+        cls,
+        data,
+    ):
         if isinstance(data, AtlanTagName):
             return data
-        if display_text := AtlanTagCache.get_name_for_id(data):
+        if display_text := AtlanClient.get_default_client_or_fail().atlan_tag_cache.get_name_for_id(
+            data
+        ):
             return AtlanTagName(display_text)
         else:
             raise ValueError(f"{data} is not a valid AtlanTag")
 
     @staticmethod
-    def json_encode_atlan_tag(atlan_tag_name: "AtlanTagName"):
-        from pyatlan.cache.atlan_tag_cache import AtlanTagCache
-
-        return AtlanTagCache.get_id_for_name(atlan_tag_name._display_text)
+    def json_encode_atlan_tag(
```

Comment:
Same comment regarding need for client as parameter

Reply:
Same comment above — I don't know how / if this will ever be sent a value by Pydantic, or will just always default to `None` (?)